src/view-impl-common.c

```diff
@@ -27,13 +27,25 @@ view_impl_move_to_back(struct view *view)
 void
 view_impl_map(struct view *view)
 {
-	if (!view->been_mapped) {
-		window_rules_apply(view, LAB_WINDOW_RULE_EVENT_ON_FIRST_MAP);
-	}
 	desktop_focus_and_activate_view(&view->server->seat, view);
 	view_move_to_front(view);
 	view_update_title(view);
 	view_update_app_id(view);
+	if (!view->been_mapped) {
+		window_rules_apply(view, LAB_WINDOW_RULE_EVENT_ON_FIRST_MAP);
+	}
+
+	/*
+	 * It's tempting to just never create the foreign-toplevel handle in the
+	 * map handlers, but the app_id/title might not have been set at that
+	 * point, so it's safer to process the property here
+	 */
+	enum property ret = window_rules_get_property(view, "skipTaskbar");
+	if (ret == LAB_PROP_TRUE) {
+		if (view->toplevel.handle) {
+			wlr_foreign_toplevel_handle_v1_destroy(view->toplevel.handle);
+		}
+	}
```

Comment:
I think this needs a `view->toplevel.handle = NULL;`, otherwise we will end up calling function on the destroyed object.

Reply:
It'd done in the destroy-handler

https://github.com/labwc/labwc/blob/f2c3f83041fce356165bd4124fd4f388bf06570d/src/foreign.c#L62

Reply:
good point.